pkg/oauth/registry/selfoauthclientauthorization/etcd/etcd.go

```diff
@@ -0,0 +1,79 @@
+package etcd
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kubeerr "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+
+	"github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/helpers"
+	"github.com/openshift/origin/pkg/util/restoptions"
+)
+
+// rest implements a RESTStorage for self oauth client authorizations against etcd
+type REST struct {
+	helpers.UIDEnforcer
+}
+
+// NewREST returns a RESTStorage object that will work against self oauth client authorizations
+func NewREST(optsGetter restoptions.Getter) (*REST, error) {
+	resource, prefix, err := helpers.GetResourceAndPrefix(optsGetter, "selfoauthclientauthorizations")
+	if err != nil {
+		return nil, fmt.Errorf("error building RESTOptions for %s store: %v", resource.String(), err)
+	}
+
+	store := &registry.Store{
+		NewFunc:     func() runtime.Object { return &api.SelfOAuthClientAuthorization{} },
```

Comment:
I would expect to have to read OAuthClientAuthorizationList from etcd and convert it yourself